packages/concerto-core/lib/introspect/recordfield.js

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Property = require('./property');
+
+// Types needed for TypeScript generation.
+/* eslint-disable no-unused-vars */
+/* istanbul ignore next */
+if (global === undefined) {
+    const ClassDeclaration = require('./classdeclaration');
+    const Validator = require('./validator');
+}
+/* eslint-enable no-unused-vars */
+
+/**
+ * Class representing the definition of a Field. A Field is owned
+ * by a ClassDeclaration and has a name, type and additional metadata
+ * (see below).
+ * @private
+ * @extends Property
+ * @see See  {@link  Property}
+ * @class
+ * @memberof module:concerto-core
+ */
+class RecordField extends Property {
+    /**
+     * Create a Field.
+     * @param {ClassDeclaration} parent - The owner of this property
+     * @param {Object} ast - The AST created by the parser
+     * @throws {IllegalModelException}
+     */
+    constructor(parent, ast) {
+        super(parent, ast);
+        this.map = new Map();
+    }
+
+    /**
+     * Process the AST and build the model
+     * @throws {IllegalModelException}
+     * @private
+     */
+    process() {
+        super.process();
```

Comment:
Properties have a `type`. What it the type for a Record? We seem to be doing something tricky here...?